test/test_cuda.py

```diff
@@ -556,241 +556,11 @@ def test_arithmetic_large_tensor(self):
         x /= 2
         self.assertEqual(x.sum(), 2**29)
 
-    def _test_broadcast(self, input):
```

Comment:
moved comm tests to a separate TestCase. Previously `test_gather` incorrectly includes a non-comm gather test.

Reply:
Would I be correct if I assume these moved tests stay intact except that they are belong to a different test class?

Reply:
Mostly, with some tests on `out=` and error message added. I'll comment to highlight the additions.